app/gui/src/presenter/graph/widgets.rs

```diff
@@ -0,0 +1,352 @@
+//! Widgets controller.
+//!
+//! The Widgets Controller is responsible for querying the language server for  information about
+//! the node's widget metadata or resolving from local cache.
+
+use crate::prelude::*;
+
+use crate::executor::global::spawn_stream_handler;
+use crate::presenter::graph::visualization::manager::Manager;
+use crate::presenter::graph::visualization::manager::Notification;
+use crate::presenter::graph::ViewNodeId;
+
+use controller::ExecutedGraph;
+use engine_protocol::language_server::SuggestionId;
+use ensogl::define_endpoints_2;
+use ide_view::graph_editor::component::node::input::widget;
+use ide_view::graph_editor::component::visualization;
+use ide_view::graph_editor::component::visualization::Metadata;
+use ide_view::graph_editor::data::enso::Code;
+use ide_view::graph_editor::WidgetUpdate;
+use ide_view::graph_editor::WidgetUpdates;
+
+
+
+/// =================
+/// === Constants ===
+/// =================
+
+const WIDGET_VISUALIZATION_MODULE: &str = "Standard.Visualization.Widgets";
+const WIDGET_VISUALIZATION_METHOD: &str = "get_full_annotations_json";
```

Comment:
these should be moved to a module where we define the language bindigns (close to AST). We should have all language bindings in one place, not scattered over the codebas.e